src/atomate2/qchem/sets/base.py

```diff
@@ -0,0 +1,157 @@
+"""Module defining base QChem input set and generator."""
+
+import os
+from pickletools import optimize
+import warnings
+from copy import deepcopy
+from dataclasses import dataclass, field
+from itertools import groupby
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+from monty.io import zopen
+from monty.serialization import loadfn
+from pkg_resources import resource_filename
+from pymatgen.core.structure import Molecule
+from pymatgen.io.qchem.inputs import QCInput
+#from pymatgen.io.qchem.utils import lower_and_check_unique
+from pymatgen.io.core import InputGenerator
+from pymatgen.io.qchem.sets import QChemDictSet
+from atomate2 import SETTINGS
+
+__author__ = "Alex Ganose, Ryan Kingsbury, Rishabh D Guha"
+__copyright__ = "Copyright 2018-2022, The Materials Project"
+__version__ = "0.1"
+
+_BASE_QCHEM_SET = loadfn(resource_filename("atomate2.qchem.sets", "BaseQchemSet.yaml"))
+
+__all__ = ["QChemInputGenerator"]
+
+@dataclass
+class QChemInputGenerator(InputGenerator):
+    """
+    A class to generate QChem input sets for different calculations
+    """
+
+    def get_input_set( 
+        self,
+        molecule: Molecule = None,
+        prev_dir: Union[str, Path] = None,
+        overwrite_inputs: dict = None
+    ) -> QChemDictSet:
+        """
+        Get a QChem Input Set as a dictionary for a molecule
+
+        Parameters
+        ----------
+        molecule
+            A Pymatgen molecule
+        prev_dir
+            A previous directory to generate the input set from
+        overwrite_inputs
+
+        Returns
+        -------
+        QchemInputSet
+            A QChem input set
+        """
+        molecule, prev_basis, prev_scf, new_geom_opt, overwrite_inputs, nbo_params = self._get_previous(
+            molecule, prev_dir
+        )
+
+        basis_set_updates = self.get_basis_set_updates(
+            molecule,
+            prev_basis=prev_basis,
+            prev_scf=prev_scf,
+            new_geom_opt=new_geom_opt,
+            overwrite_inputs=overwrite_inputs,
+            nbo_params=nbo_params,
+        )
+
+        scf_algorithm_updates = self.get_scf_algorithm_updates(
+            molecule,
+            prev_basis=prev_basis,
+            prev_scf=prev_scf,
+            new_geom_opt=new_geom_opt,
+            overwrite_inputs=overwrite_inputs,
+            nbo_params=nbo_params,
+        )
+
+        basis_set = self._basis_set(basis_set_updates)
+        scf_algorithm = self._get_scf_algorithm(scf_algorithm_updates)
+
+
```

Comment:
The way I envisioned implementing an `InputGenerator` is that any of the core calculation settings, such as basis_set and scf_algorithm, are set as class attributes on instantiation, e.g.

```QChemInputGenerator(basis_set = 'def2-svpd')``` etc. 

Basically, any argument that needs to be passed to `InputSet` _other than` the strutcure. Then_ the `get_input_set` passes the class attrs, plus the `Molecule`, to the input set init method.

I could be missing something, but for me, having all these extra methods makes the code harder to understand, when really their only purpose is to set some kwargs in the InputSet constructor.